RecoTracker/MkFitCore/src/PropagationMPlex.cc

```diff
@@ -655,9 +655,11 @@ namespace mkfit {
           hitsRl(n, 0, 0) = mat.radl;
           hitsXi(n, 0, 0) = mat.bbxi;
         }
-        const float zout = msZ.constAt(n, 0, 0);
-        const float zin = inPar.constAt(n, 2, 0);
-        propSign(n, 0, 0) = (std::abs(zout) > std::abs(zin) ? 1. : -1.);
+        if (n < N_proc) {
+          const float zout = msZ.constAt(n, 0, 0);
+          const float zin = inPar.constAt(n, 2, 0);
+          propSign(n, 0, 0) = (std::abs(zout) > std::abs(zin) ? 1. : -1.);
```

Comment:
this will conflict with the prop to plane PR now open in cmssw or equivalently in trackreco #136 
But then we have an unresolved difference there .
 should we just merge this with #136  or anticipate a rebase?

Reply:
Probably best to wait for #136 to be merged and then I'll rebase